### PR TITLE
QAG2-105 Create Base functionality

### PR DESCRIPTION
**Pull request by EmmLazyan**
Create Base functionality.

### CODE REVIEW

File: pages_/basePage.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+            exit(1)
+
+    def _click(self, webElement):
+        webElement.click()
+
+    def _fill_field(self, webElement, text):
+        webElement.clear()
+        webElement.send_keys(text)
+
+    def _get_title(self):
```
Comment (satikmkhitaryan) [inline pages_/basePage.py:26]: input 
        logger("INFO", f"The title is found successfully: {self.driver.title}")

File: pages_/basePage.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        return self.driver.title
+
+    def _get_element_text(self):
```
Comment (satikmkhitaryan) [inline pages_/basePage.py]: input
 logger("INFO", f"Text is founded: {webElement.text}")

File: pages_/basePage.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+class BasePage():
+    def _init_(self, driver: webdriver.Chrome):
+        self.driver = driver
+
+    def _find_element(self, by, value):
+        try:
+            element = WebDriverWait(self.driver, 10).until(EC.visibility_of_element_located((by, value)))
+            return element
+        except:
+            print("Error: Element not Found")
```
Comment (satikmkhitaryan) [inline pages_/basePage.py]: Instead print input
logger("ERROR", "Element not found"

File: pages_/basePage.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
+from selenium import webdriver
+from selenium.webdriver.support.ui import WebDriverWait
+from selenium.webdriver.support import expected_conditions as EC
+
```
Comment (satikmkhitaryan) [inline pages_/basePage.py:5]: from common_.utilities_.customLogger import *

Comment (EmmLazyan) [reply pages_/basePage.py:5]: The changes are complated.

File: pages_/basePage.py [hunk 5]
```diff
@@ -24,6 +25,8 @@ def _fill_field(self, webElement, text):
 
     def _get_title(self):
         return self.driver.title
+        logger("INFO", f"The title is found successfully: {self.driver.title}")
```
Comment (satikmkhitaryan) [inline pages_/basePage.py:27]: Input logger ... before returm  self.driver.title

File: pages_/basePage.py [hunk 6]
```diff
@@ -24,6 +25,8 @@ def _fill_field(self, webElement, text):
...
 
     def _get_element_text(self):
         return self.element.text
+        logger("INFO", f"Text is founded: {webElement.text}")
```
Comment (satikmkhitaryan) [inline pages_/basePage.py:31]: Input logger ... before returm self.driver.title

Comment (EmmLazyan) [reply pages_/basePage.py:31]: _gat_title and _find_element methods are changed.

**Review by satikmkhitaryan — APPROVED** [review]

**Pull request merged**